dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,682 @@
+    }
+
+    public  String[] getAccessFilterArray(final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        final String accessFilterValues = this.samlConfigurationService.getConfigAsString(identityProviderConfiguration, SamlName.DOT_SAML_ACCESS_FILTER_VALUES);
+
+        return UtilMethods.isSet( accessFilterValues ) ?
+                accessFilterValues.split( "," ) : null;
+    }
+
+    protected boolean isByPass(final HttpServletRequest request, final HttpSession session) {
+
+        String byPass = request.getParameter(BY_PASS_KEY);
+
+        if (null != session) {
+            if (null != byPass) {
+
+                session.setAttribute(BY_PASS_KEY, byPass);
+            } else {
+                if (this.isNotLogged(request, session)) {
+
+                    byPass = (String) session.getAttribute(BY_PASS_KEY);
+                } else if (null != session.getAttribute(BY_PASS_KEY)) {
+
+                    session.removeAttribute(BY_PASS_KEY);
+                }
+            }
+        }
+
+        return BY_PASS_VALUE.equalsIgnoreCase(byPass);
+    }
+
+    /**
+     * Return true if the user is not logged. Work for FE and BE
+     *
+     * @param request
+     *            {@link HttpServletRequest}
+     * @param session
+     *            {@link HttpSession}
+     * @return boolean
+     */
+    protected boolean isNotLogged(final HttpServletRequest request, final HttpSession session) {
+
+        boolean isNotLogged = true;
+        boolean isBackend   = this.isBackEndAdmin(request, request.getRequestURI());
+        
+        try {
+            isNotLogged = isBackend? !this.userWebAPI.isLoggedToBackend(request)
+                    : null == this.userWebAPI.getLoggedInFrontendUser(request);
+
+            Logger.debug(this, "Trying to go to back-end login? " + isBackend
+                    + ", Is user NOT logged in? " + isNotLogged);
+        } catch (PortalException | SystemException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            isNotLogged = true;
+        }
+
+        return isNotLogged;
+    }
+
+    /**
+     * Determines whether the user in the {@link HttpServletRequest} object or the incoming URI belong to the
+     * dotCMS back-end login mechanism or not.
+     *
+     * @param request The {@link HttpServletRequest} request
+     * @param uri     The incoming URI for login.
+     *
+     * @return If the user or its URI can be associated to the dotCMS back-end login, returns {@code true}. Otherwise,
+     * returns {@code false}.
+     */
+    protected boolean isBackEndAdmin(final HttpServletRequest request, final String uri) {
+
+        return PageMode.get(request).isAdmin || this.isBackEndLoginPage(uri);
+    }
+
+    protected boolean isFrontEndLoginPage(final String uri){
+
+    	return uri.startsWith("/dotCMS/login") || uri.startsWith("/application/login");
+    }
+
+    public boolean isLogoutRequest(final String requestURI, final String[] logoutPathArray) {
```

Comment:
Codacy found an issue: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)